src/nvim/edit.c

```diff
@@ -4160,25 +4162,27 @@ static int ins_compl_get_exp(pos_T *ini)
               STRNCPY(IObuff, ptr, len);
               ptr = ml_get_buf(ins_buf, pos->lnum + 1, false);
               tmp_ptr = ptr = skipwhite(ptr);
-              /* Find start of next word. */
+              // Find start of next word.
               tmp_ptr = find_word_start(tmp_ptr);
-              /* Find end of next word. */
+              // Find end of next word.
               tmp_ptr = find_word_end(tmp_ptr);
               if (tmp_ptr > ptr) {
                 if (*ptr != ')' && IObuff[len - 1] != TAB) {
-                  if (IObuff[len - 1] != ' ')
+                  if (IObuff[len - 1] != ' ') {
                     IObuff[len++] = ' ';
-                  /* IObuf =~ "\k.* ", thus len >= 2 */
+                  }
+                  // IObuf =~ "\k.* ", thus len >= 2
                   if (p_js
                       && (IObuff[len - 2] == '.'
                           || IObuff[len - 2] == '?'
                           || IObuff[len - 2] == '!')) {
                     IObuff[len++] = ' ';
                   }
                 }
-                /* copy as much as possible of the new word */
-                if (tmp_ptr - ptr >= IOSIZE - len)
+                // copy as much as possible of the new word
+                if (tmp_ptr - ptr >= IOSIZE - len) {
                   tmp_ptr = ptr + IOSIZE - len - 1;
+                }
                 STRNCPY(IObuff + len, ptr, tmp_ptr - ptr);
```

Comment:
Linter emits warning here
> Use xstrlcpy or snprintf instead of STRNCPY (unless this is from Vim)

but both options lead to compile warnings, so I left this as is.